training/src/app/modules/chuck-norris/components/search/search.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, Output, EventEmitter, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from "@angular/forms";
+import { debounceTime, tap } from "rxjs";
+
+@Component({
+	selector: 'app-search',
+	templateUrl: './search.component.html',
+	styleUrls: ['./search.component.scss']
+})
+export class SearchComponent implements OnInit {
+
+	searchForm: FormGroup;
+	public searchInput: string = '';
+	@Output() private generateJokesOnClickMode: EventEmitter<string> = new EventEmitter<string>();
+	@Output() private generateJokesOnSearchMode: EventEmitter<string> = new EventEmitter<string>();
+
+	constructor(private fb: FormBuilder) {
+	}
+
+	ngOnInit(): void {
+		this.searchForm = this.fb.group({
+			searchKey: [
+				'', [
+					Validators.required,
+					Validators.minLength(3),
+					Validators.maxLength(120)
+				]
+			],
+		});
+		this.searchKey?.valueChanges.pipe(
+			tap(search => {
+				this.generateJokesOnClickMode.emit(search);
+				}
+			),
+			debounceTime(1000)
+		).subscribe(value => this.setItems(value))
```

Comment:
add typing

Reply:
we learned that Angular wont unsubscribe from observables for us, you need to do it